doc/qbk/02_9_conversion.qbk

```diff
@@ -13,101 +13,303 @@
+```
+void tag_invoke( value_from_tag, value&, T );
+```
 
-* A public constructor `T::T( value const& )`,
+Likewise, the overloads of `tag_invoke` called by __value_to__
+take the form:
 
-The library provides built-in support for converting between a __value__
-and the type `T` when:
+```
+T tag_invoke( value_to_tag< T >, const value& );
+```
 
-* `std::is_constructible< value, T, storage_ptr >::value == true`, or
+In both cases, overloads for user-provided types can be implemented:
 
-* `T` is a container meeting a built-in list of supported generic requirements.
+[snippet_conv_1]
 
-When more than one conversion is available for a given type, one
-will be chosen according to a defined set of priorities.
+Since the type being converted to/from is embedded into the
+parameter-type-list of the function, user-provided overloads
+are visible to argument-dependent lookup and will be considered
+when a conversion is performed. Consider the following:
 
+[snippet_conv_adl_1]
 
+When __value_from__ is called, the `tag_invoke` function template
+will be found by argument-dependent lookup and used to perform the conversion:
 
-[heading Member Functions]
+[snippet_conv_adl_2]
 
-The example code that follows uses the user-defined type `customer`,
-declared thusly:
+In addition to user-provided overloads of `tag_invoke`, the library
+will add its own function to the overload set when certain constraints
+are satisfied. The library provided overloads have no special
+prioritization over those provided by the user, so care should be taken to
+avoid writing ambiguous declarations:
 
-[snippet_exchange_1]
+[snippet_conv_ambig_1]
 
-Since the author has control of the type, it uses member functions
-to enable conversions with JSON. An implementation of `to_json` may
-look like this:
+Upon calling this function, overload resolution will fail because
+the library already provides an overload for floating-point types:
 
-[snippet_exchange_2]
+[snippet_conv_ambig_2]
 
-Turning a JSON value back into a customer is a bit more complicated,
-as it must handle the case where the value does not have the necessary
-keys, or if the values have the wrong kind. The implementation of
-`from_json` throws an exception when the JSON value does not precisely
-match the expected schema for customer:
+Library-provided overloads of `tag_invoke` come in two flavors: those that
+convert between JSON types (known as ['built-in conversions]),
+and those that convert to/from container and string types
+(known as ['generic conversions]). Generic conversions offer convenience
+by eliminating the need to write repetitive overloads for types
+that model common C++ concepts (e.g. sequence containers, associative containers,
+tuples, and strings).
 
-[snippet_exchange_3]
+[snippet_conv_generic]
 
-When conversion between `T` and __value__ is defined in both directions,
-round trips are possible:
+[heading Converting to __value__]
```

Comment:
Don't generate links headings, use this instead:
```
[heading Converting to json::value]
```

Quickbook can get weird with formatting otherwise.